### PR TITLE
Avoid error in EnggFocus by not normalising by current if run has zero proton charge

### PR DESCRIPTION
**Pull request by RichardWaiteSTFC**
**Description of work.**

Small bug fix checks whether a run has proton charge before running NormaliseByCurrent - which would throw a runtime error. This behaviour ahs been requested such that when the algorithm is used in the EngDiff UI workflow on multiple runs it doesn't stop on an empty run.

**To test:**
(1) Load this calibraiton in the calibration tab of the EnggDiff UI
[ENGINX_307521_305738_all_banks.zip](https://github.com/mantidproject/mantid/files/6377292/ENGINX_307521_305738_all_banks.zip)
(2) Go to the Focus tab, type 326016 in the sample run box and press Focus
(3) Focussing should execute without error (note the focussed workspace is full of zeros as there was no proton charge)

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/test/python/plugins/algorithms/EnggFocusTest.py [hunk 1]
```diff
@@ -226,6 +226,18 @@ def test_runs_ok_bank_south(self):
...
+        """
+        Run focusing on one bank but on a ws with no proton charge
+        """
+        # remove proton charge log (getProtonCharge will then return 0)
+        ws = CloneWorkspace(self.__class__._data_ws)  # so don't interfere with other tests
+        DeleteLog(Workspace=ws, Name='gd_prtn_chrg')
+        out_bank_south = EnggFocus(InputWorkspace=ws,
+                                   VanIntegrationWorkspace=self.__class__._van_integ_tbl,
+                                   VanCurvesWorkspace=self.__class__._van_curves_ws,
+                                   Bank='South', OutputWorkspace='out_bank_south')  # test passes if no RuntimeError
```
Comment (thomashampson) [inline Framework/PythonInterface/test/python/plugins/algorithms/EnggFocusTest.py:239]: Could move the body of this to a new function, e.g. "run_zero_proton_charge(self)", and replace the existing function with:
```
def test_no_error_when_zero_proton_charge(self):
    assertRaisesNothing(self, run_zero_proton_charge)
```
which would require `from testhelpers import assertRaisesNothing`

**Review by thomashampson — CHANGES_REQUESTED** [review]
Works as expected. One minor suggestion to slightly refactor the test case.

File: Framework/PythonInterface/test/python/plugins/algorithms/EnggFocusTest.py (hunk 1 above)
Comment (RichardWaiteSTFC) [reply Framework/PythonInterface/test/python/plugins/algorithms/EnggFocusTest.py:239]: Thanks for this suggestion, as discussed offline I was looking for something like this!

**Review by thomashampson — APPROVED** [review]

**Review by DavidFair — APPROVED** [review]
LGTM and I didn't know about assertRaisesNothing, I previously let unittest catch them so I've learnt something too :+1:

**Pull request merged**